internal/controlplane/handlers_profile.go

```diff
@@ -364,28 +330,47 @@ func (s *Server) GetProfileById(ctx context.Context,
 		return nil, util.UserVisibleError(codes.InvalidArgument, "invalid profile ID")
 	}
 
-	profiles, err := s.store.GetProfileByProjectAndID(ctx, db.GetProfileByProjectAndIDParams{
+	prof, err := getProfilePBFromDB(ctx, parsedProfileID, entityCtx, s.store)
+	if err != nil {
+		if errors.Is(err, sql.ErrNoRows) || strings.Contains(err.Error(), "not found") {
```

Comment:
when do we get the second case? Wouldn't it be better to catch that issue by Postgres error code if that error comes directly from Postgres?

Reply:
the "not found" is not a postgres error. I just didn't create a specialized error for this :confused: